src/BlockTypes/AddToCartForm.php

```diff
@@ -36,30 +36,496 @@ protected function render( $attributes, $content, $block ) {
+	 *
+	 * @param \WC_Product $product Product object.
+	 * @param array       $attributes Array of attributes.
+	 * @param array       $attribute_keys Array of attribute keys.
+	 * @param array       $available_variations Array of available variations.
+	 * @param string      $variations_attr HTML encoded string of variations.
+	 *
+	 * @return string
+	 */
+	protected function variable_product_form( $product, $attributes, $attribute_keys, $available_variations, $variations_attr ) {
+		ob_start();
+		/**
+		 * Hook: woocommerce_before_add_to_cart_form.
+		 *
+		 * @since TBD
+		 */
+		do_action( 'woocommerce_before_add_to_cart_form' );
+
+		$product_id = $product->get_id();
+		?>
+		<form class="variations_form cart" action="<?php echo esc_url( $product->get_permalink() ); ?>" method="post" enctype='multipart/form-data' data-product_id="<?php echo absint( $product_id ); ?>" data-product_variations="<?php echo wc_esc_json( $variations_attr ); ?>">
+		<?php
+		/**
+		 * Hook: woocommerce_before_variations_form.
+		 *
+		 * @since TBD
+		 */
+		do_action( 'woocommerce_before_variations_form' );
+		?>
+
+		<?php if ( empty( $available_variations ) && false !== $available_variations ) : ?>
+			<p class="stock out-of-stock"><?php echo esc_html__( 'This product is currently out of stock and unavailable.', 'woo-gutenberg-products-block' ); ?></p>
+		<?php else : ?>
+			<table class="variations" role="presentation">
+				<tbody>
+				<?php foreach ( $attributes as $attribute_name => $options ) : ?>
+					<tr>
+						<th class="label"><label for="<?php echo esc_attr( sanitize_title( $attribute_name ) ); ?>"><?php echo esc_html( wc_attribute_label( $attribute_name ) ); ?></label></th>
+						<td class="value">
+							<?php
+							wc_dropdown_variation_attribute_options(
+								array(
+									'options'   => $options,
+									'attribute' => $attribute_name,
+									'product'   => $product,
+								)
+							);
+							echo end( $attribute_keys ) === $attribute_name ? wp_kses_post( '<a class="reset_variations" href="#">' . esc_html__( 'Clear', 'woo-gutenberg-products-block' ) . '</a>' ) : '';
+							?>
+						</td>
+					</tr>
+				<?php endforeach; ?>
+				</tbody>
+			</table>
+			<?php
+			/**
+			 * Hook: woocommerce_after_variations_table.
+			 *
+			 * @since TBD
+			 */
+			do_action( 'woocommerce_after_variations_table' );
+			?>
+			<div class="single_variation_wrap">
+				<?php
+				/**
+				 * Hook: woocommerce_before_single_variation.
+				 *
+				 * @since TBD
+				 */
+				do_action( 'woocommerce_before_single_variation' );
+				?>
+				<div class="woocommerce-variation single_variation"></div>
+				<div class="woocommerce-variation-add-to-cart variations_button">
+					<?php
+					$this->get_quantity_input( $product );
+					$this->add_to_cart_button( $product );
+					$this->variable_inputs( $product_id );
+					?>
+				</div>
+				<?php
+				/**
+				 * Hook: woocommerce_after_single_variation.
+				 *
+				 * @since TBD
```

Comment:
If you use the version placeholder, the release script will automatically add the correct version string.
```suggestion
				 * @since $:VID:$
```
This could be updated in multiple places.

Reply:
Nice! I wasn't aware of this script, TIL!